File: packages/python/palpluss/types/stk.py
from __future__ import annotations

from typing_extensions import TypedDict


class StkInitiateResponse(TypedDict):
    transactionId: str
    tenantId: str
    channelId: str | None
    type: str
    status: str
    amount: float
    currency: str
    phone: str
    accountReference: str
    transactionDesc: str
    providerRequestId: str
    providerCheckoutId: str
    resultCode: str
    resultDescription: str
    createdAt: str
    updatedAt: str

File: packages/typescript/src/types/stk.ts
export interface StkInitiateRequest {
  amount: number;
  phone: string;
  accountReference?: string;
  transactionDesc?: string;
  channelId?: string;
  callbackUrl?: string;
  credential_id?: string;
}

export interface StkInitiateResponse {
  transactionId: string;
  tenantId: string;
  channelId: string | null;
  type: 'STK';
  status: 'PENDING';
  amount: number;
  currency: string;
  phone: string;
  accountReference: string;
  transactionDesc: string;
  providerRequestId: string;
  providerCheckoutId: string;
  resultCode: string;
  resultDescription: string;
  createdAt: string;
  updatedAt: string;
}
